pkg/hubble/filters/tcp_test.go

```diff
@@ -0,0 +1,117 @@
+
+package filters
+
+import (
+	"context"
+	"testing"
+
+	flowpb "github.com/cilium/cilium/api/v1/flow"
+	v1 "github.com/cilium/cilium/pkg/hubble/api/v1"
+)
+
+func TestFlowTCPFilter(t *testing.T) {
+	var argsevent *v1.Event
+	testflags := []struct {
+		name   string
+		argsf  []*flowpb.FlowFilter
+		argsev *flowpb.TCPFlags
+		want   bool
+	}{
+		{
+			name:   "filterSYNACK___eventSYNACK",
+			argsf:  []*flowpb.FlowFilter{{TcpFlags: []*flowpb.TCPFlags{{SYN: true, ACK: true}}}},
+			argsev: &flowpb.TCPFlags{SYN: true, ACK: true},
+			want:   true,
+		},
+		{
+			name:   "filterSYNACK___eventSYN",
+			argsf:  []*flowpb.FlowFilter{{TcpFlags: []*flowpb.TCPFlags{{SYN: true, ACK: true}}}},
+			argsev: &flowpb.TCPFlags{SYN: true},
+			want:   false,
+		},
+		{
+			name: "filterSYN_OR_filterACK___eventSYN",
+			argsf: []*flowpb.FlowFilter{{TcpFlags: []*flowpb.TCPFlags{{SYN: true}}},
+				{TcpFlags: []*flowpb.TCPFlags{{ACK: true}}}},
+			argsev: &flowpb.TCPFlags{SYN: true},
+			want:   true,
+		},
+		{
+			name:   "filterSYN___eventSYN",
+			argsf:  []*flowpb.FlowFilter{{TcpFlags: []*flowpb.TCPFlags{{SYN: true}}}},
+			argsev: &flowpb.TCPFlags{SYN: true},
+			want:   true,
+		},
+		{
+			name:   "filterSYN___eventSYNACK",
+			argsf:  []*flowpb.FlowFilter{{TcpFlags: []*flowpb.TCPFlags{{SYN: true}}}},
+			argsev: &flowpb.TCPFlags{SYN: true, ACK: true},
+			want:   true,
+		},
+		{
+			name:   "filterFIN___eventRST",
+			argsf:  []*flowpb.FlowFilter{{TcpFlags: []*flowpb.TCPFlags{{FIN: true}}}},
+			argsev: &flowpb.TCPFlags{RST: true},
+			want:   false,
+		},
+		{
+			name:   "filterSYN___eventPSH",
+			argsf:  []*flowpb.FlowFilter{{TcpFlags: []*flowpb.TCPFlags{{SYN: true}}}},
+			argsev: &flowpb.TCPFlags{PSH: true},
+			want:   false,
+		},
+		{
+			name:   "filterURG___eventPSH",
+			argsf:  []*flowpb.FlowFilter{{TcpFlags: []*flowpb.TCPFlags{{URG: true}}}},
+			argsev: &flowpb.TCPFlags{PSH: true},
+			want:   false,
+		},
+		{
+			name:   "filterRST___eventRST",
+			argsf:  []*flowpb.FlowFilter{{TcpFlags: []*flowpb.TCPFlags{{RST: true}}}},
+			argsev: &flowpb.TCPFlags{RST: true},
+			want:   true,
+		},
+		{
+			name:   "filterFIN___eventFIN",
+			argsf:  []*flowpb.FlowFilter{{TcpFlags: []*flowpb.TCPFlags{{FIN: true}}}},
+			argsev: &flowpb.TCPFlags{FIN: true},
+			want:   true,
+		},
+		{
+			name:   "filterPSH___eventPSHACK",
+			argsf:  []*flowpb.FlowFilter{{TcpFlags: []*flowpb.TCPFlags{{PSH: true}}}},
+			argsev: &flowpb.TCPFlags{PSH: true, ACK: true},
+			want:   true,
+		},
+	}
+
+	for _, tt := range testflags {
+		argsevent = &v1.Event{Event: &flowpb.Flow{
+			L4: &flowpb.Layer4{Protocol: &flowpb.Layer4_TCP{TCP: &flowpb.TCP{Flags: tt.argsev}}}}}
+		t.Run(tt.name, func(t *testing.T) {
+			fl, err := BuildFilterList(context.Background(), tt.argsf, []OnBuildFilter{&TCPFilter{}})
+			if err != nil {
+				return
+			}
+			if got := fl.MatchOne(argsevent); got != tt.want {
+				t.Errorf("\"%s\" got %v, want %v", tt.name, got, tt.want)
```

Comment:
```
`"%s" got %v, want %v`
```
avoids escaping

Also quoting it is not necessary. `%s: got %v, want %v"` is probably sufficient

Reply:
Fixed.